### PR TITLE
[hybrid-aot] Fix compilation and tests compilation for MOBILE profile

### PR DESCRIPTION
**Pull request by ludovic-henry**
cc @kumpera @vargaz @alexanderkyte

### CODE REVIEW

**Comment by marek-safar** [comment]
Why are you using mobile naming for hybrid AOT?

**Comment by ludovic-henry** [comment]
@marek-safar it just happen that we use the `mobile` profile for testing hybrid AOT. So most of this PR is about fixing the mobile profile build so that we can test it on desktop.

**Comment by marek-safar** [comment]
@ludovic-henry and I don't like that approach. You should not hijack existing name to do something quite different. Either rename mobile to hybrid_aot or something like that or introduce a new profile

**Comment by ludovic-henry** [comment]
@marek-safar it's the same approach as using mobile_static for testing on desktop, even if we know the the only usage of mobile_static in monotouch which is very close to it. Also, I initially looked into creating a new profile for that, and it seemed like it's way to complicated for not such a big gain. Finally, we came to this use with @kumpera as nothing uses mobile: monodroid and xammac are different profiles, which defines mostly the same thing, but different anyway.

**Comment by marek-safar** [comment]
and I don't want to repeat the mistake, so please name the profile appropriately. As I wrote I don't mind renaming mobile profile as it's unused anyway to something like `mobile_aot_hybrid`

File: mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -10436,7 +10438,7 @@ mono_compile_assembly (MonoAssembly *ass, guint32 opts, const char *aot_options)
 	}
 
 #if defined(MONO_ARCH_GSHAREDVT_SUPPORTED)
-	if (acfg->aot_opts.llvm_only || mono_aot_mode_is_full (&acfg->aot_opts)) {
+	if (acfg->aot_opts.llvm_only || mono_aot_mode_is_full (&acfg->aot_opts) || mono_aot_mode_is_hybrid (&acfg->aot_opts)) {
```
Comment (xmcclure) [inline mono/mini/aot-compiler.c]: Would it make sense to bottleneck (mono_aot_mode_is_full (&acfg->aot_opts) || mono_aot_mode_is_hybrid (&acfg->aot_opts)) on a no_il_available() or gsharedvt_enabled() or something? I notice you use those checks in two places, and it is a little opaque to me looking through this code to understand why the choice was made between which of (full aot, llvm, hybrid) is checked for any particular if clause.

File: mcs/class/corlib/ReferenceSources/RuntimeType.cs [hunk 2]
```diff
@@ -32,7 +32,7 @@
 using System.Collections.Generic;
 using System.Runtime.InteropServices;
 using System.Globalization;
-#if MONO_COM
+#if FEATURE_COMINTEROP
```
Comment (xmcclure) [inline mcs/class/corlib/ReferenceSources/RuntimeType.cs]: I think we are trying to make sure some MONO symbol wraps any change we have made to the ReferenceSources/ directory. So maybe this should be #if MONO && FEATURE_COMINTEROP

File: mcs/class/referencesource/mscorlib/system/activator.cs [hunk 3]
```diff
@@ -608,7 +608,7 @@ static public T CreateInstance<T>()
                                                                    activationAttributes,
                                                                    null);
         }
-#if FEATURE_COMINTEROP || MONO_COM || MOBILE_LEGACY
+#if FEATURE_COMINTEROP || MOBILE_LEGACY
```
Comment (xmcclure) [inline mcs/class/referencesource/mscorlib/system/activator.cs]: See comment re #if MONO

File: mcs/class/referencesource/mscorlib/system/io/textwriter.cs [hunk 4]
```diff
@@ -39,7 +39,7 @@ namespace System.IO {
     // There are methods on the Stream class for writing bytes. 
     [Serializable]
     [ComVisible(true)]
-#if FEATURE_REMOTING
+#if FEATURE_REMOTING || MOBILE_LEGACY
```
Comment (xmcclure) [inline mcs/class/referencesource/mscorlib/system/io/textwriter.cs]: See comment re #if MONO

File: mcs/class/referencesource/mscorlib/system/type.cs [hunk 5]
```diff
@@ -155,7 +155,7 @@ public static Type ReflectionOnlyGetType(String typeName, bool throwIfNotFound,
         public virtual Type MakeArrayType() { throw new NotSupportedException(); }
         public virtual Type MakeArrayType(int rank) { throw new NotSupportedException(); }
 
-#if FEATURE_COMINTEROP || MONO_COM || MOBILE_LEGACY
+#if FEATURE_COMINTEROP || MOBILE_LEGACY
```
Comment (xmcclure) [inline mcs/class/referencesource/mscorlib/system/type.cs]: See comment re #if MONO

File: configure.ac [hunk 6]
```diff
@@ -765,49 +766,29 @@ enable_llvm_default=no
...
 
+with_bitcode_default=no
+with_cooperative_gc_default=no
+with_mobile_default=no
+with_mobile_static_default=no
+with_monodroid_default=no
+with_monotouch_default=no
+with_monotouch_tv_default=no
+with_monotouch_watch_default=no
+with_profile4_x_default=no
```
Comment (ludovic-henry) [inline configure.ac]: This is a test for github required comments

see https://help.github.com/articles/about-pull-request-reviews/

File: configure.ac [hunk 7]
```diff
@@ -765,49 +766,29 @@ enable_llvm_default=no
...
    DISABLE_MCS_DOCS_default=yes
-   with_profile4_x_default=no
-   with_monodroid_default=no
-   with_monotouch_default=no
-   with_monotouch_watch_default=no
-   with_monotouch_tv_default=no
-   with_xammac_default=no
-   with_mobile_static_default=no
-   with_bitcode_default=no
-   with_cooperative_gc_default=no
```
Comment (ludovic-henry) [inline configure.ac]: test2

**Pull request closed without merging**